### ISSUE TITLE
--build-solutions requires editor window when .mono/assemblies is not present

### ISSUE THREAD

**Issue by KellyThomas** [godotengine/godot#18799]
**Godot version:**
godot mono 3.0.3-rc1 custom build

**OS/device including version:**
windows 7 

**Issue description:**

Godot is only able to build a project in `--no-window` mode when the `.mono\assemblies` directory has already been populated with the appropriate dlls.  If these dlls are not present then the build will fail.

This constraint does not apply if the editor window is displayed. 

**Steps to reproduce:**

If I have a project directory freshly cloned from source control and without the `.mono\assemblies` directory and I call godot with `-q --no-window --build-solutions` then assemblies are neither found nor created  and the build fails with this log:

```
C:\Apps\godot>godot.windows.opt.tools.64.mono.exe -v -q --no-window --build-solutions --path C:\projects\example
Using GLES3 video driver
OpenGL ES 3.0 Renderer: GeForce 405/PCIe/SSE2
WASAPI: detected 2 channels
WASAPI: audio buffer frames: 933 calculated latency: 21ms
Mono: Initializing module...
Mono: Logfile is C:\Users\Bill\AppData\Roaming/Godot/mono/mono_logs/1526063901.txt
Mono: Runtime initialized
Mono: Loading assembly mscorlib...
Mono: Assembly mscorlib loaded from path: C:\Apps\godot/mscorlib.dll
Mono: Loading tools domain...
Mono: Loading assembly GodotSharpTools...
Mono: Assembly GodotSharpTools loaded from path: C:\Apps\godot/GodotSharpTools.dll
Mono: Loading scripts domain...
Mono: Debugger wait timeout
Mono: Loading assembly GodotSharp...
Mono: Failed to load Core API assembly
Mono: INITIALIZED
CORE API HASH: -5348773433895422571
EDITOR API HASH: -797084308862332505
load resource: res://default_env.tres
load resource: res://main.tscn
Mono: Unloading scripts domain...
Mono: Runtime cleanup...
debugger-agent: Unable to listen on 1020
```

However if instead I call godot with `-q -e --build-solutions` then the assemblies are created  and the build completes with this log:

```
C:\Apps\godot>godot.windows.opt.tools.64.mono.exe -v -q -e --build-solutions --path C:\projects\example
Using GLES3 video driver
OpenGL ES 3.0 Renderer: GeForce 405/PCIe/SSE2
WASAPI: detected 2 channels
WASAPI: audio buffer frames: 933 calculated latency: 21ms
Mono: Initializing module...
Mono: Logfile is C:\Users\Bill\AppData\Roaming/Godot/mono/mono_logs/1526062617.txt
Mono: Runtime initialized
Mono: Loading assembly mscorlib...
Mono: Assembly mscorlib loaded from path: C:\Apps\godot/mscorlib.dll
Mono: Loading tools domain...
Mono: Loading assembly GodotSharpTools...
Mono: Assembly GodotSharpTools loaded from path: C:\Apps\godot/GodotSharpTools.dll
Mono: Loading scripts domain...
Mono: Debugger wait timeout
Mono: Loading assembly GodotSharp...
Mono: Failed to load Core API assembly
Mono: INITIALIZED
CORE API HASH: -5348773433895422571
EDITOR API HASH: -797084308862332505
load resource: res://default_env.tres
load resource: C:\Users/Bill/AppData/Roaming/Godot/editor_settings-3.tres
EditorSettings: Load OK!
load resource: res://default_env.tres
load resource: res://main.tscn
EditorSettings Save OK!
Audio Driver WASAPI average latency: 22ms (frame=933)
EditorSettings Save OK!
Mono: Unloading scripts domain...
Mono: Runtime cleanup...
```

**Comment by exts** [comment]
Looks like this completely fails in general on windows 10 even if the .mono folder exists

**Comment by NathanWarden** [comment]
I think that makes sense. I wrote the code under Linux where I don't think the -no-window mode does anything (maybe I'm wrong). So, this code is maybe only called when all progress bars have finished? That's my best guess. I just started a new job and moved into a new place with my family which eats up most of my time, so I likely won't get to this anytime soon. If anyone else wants to fix this would be awesome. :)

**Comment by neikeq** [comment]
Shouldn't this be with both `--no-windows` and `-e`?

**Comment by KellyThomas** [comment]
In my experience (and I could be doing something wrong)  `-e` overrules `--no-window`.

**Comment by neikeq** [comment]
But the editor only runs if you pass `-e`, right? How is `--build-solutions` supposed to work without the editor?:

https://github.com/godotengine/godot/blob/07d4cd7f6d833332b1bc3d2025d222778624ce17/main/main.cpp#L1881-L1886

**Comment by KellyThomas** [comment]
OK, that certainly explains why I couldn't get it working from a cold start.

I guess I was mistaken in thinking it was performing a `--no-window` build correctly at any point.

**Comment by neikeq** [comment]
The example in #16098 uses `-e --build-solutions`.

**Comment by akien-mga** [comment]
Fixed by #19589.
